### ISSUE TITLE
[DOXIA-553] Replace pegdown due to deprecation and end of life notice

### ISSUE THREAD

**Issue by jira-importer** [apache/maven-doxia#531]
**[Marc Campbell](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=moonrocket)** opened **[DOXIA-553](https://issues.apache.org/jira/browse/DOXIA-553?redirect=false)** and commented

On 2016.12.14, [`sirthias/pegdown`](https://github.com/sirthias/pegdown/) posted a deprecation notice that "pegdown has reached its end of life."

The pegdown notice also gives reasons to consider [`@vsch's`](https://github.com/vsch) [`flexmark-java`](https://github.com/vsch/flexmark-java) as a replacement.

See:
https://github.com/sirthias/pegdown
https://github.com/vsch/flexmark-java

---

**Issue Links:**
- [DOXIA-554](https://issues.apache.org/jira/browse/DOXIA-554) Parsing Markdown documents can hang site generation: switch parser from Pegdown to Flexmark

1 votes, 3 watchers

**Comment by jira-importer** [comment]
**[Vladimir Schneider](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=vsch)** commented

I am willing to provide a PR. I am setting up my dev environment now so I can make the needed changes, test and generate a PR.

I migrated https://github.com/walokra/markdown-page-generator-plugin to flexmark-java https://github.com/vsch/markdown-page-generator-plugin without much trouble, since the library was developed as a replacement for pegdown in my [IntelliJ plugin project](https://github.com/vsch/idea-multimarkdown). I don't expect any issues other than my lack of DOXIA and Maven plugin experience.

**Comment by jira-importer** [comment]
**[Marc Campbell](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=moonrocket)** commented

As an FYI...

About a month ago, I put together a maven project to evaluate 3 maven plugins: doxia-module-markdown (pegdown), walokra/markdown-page-generator-plugin (pegdown) and vsch/markdown-page-generator-plugin (flexmark-java).

https://github.com/marc-medley/004.55.markdown-maven-plugin-eval

Subsequent to the evaluation, I added vsch/markdown-page-generator-plugin (flexmark-java) to projects I'm working on.  The flexmark-java plugin has been reliable.

So, once there is a doxia-module-markdown with flexmark-java, I can help with some testing as a user of such plugin.

**Comment by jira-importer** [comment]
**[Vladimir Schneider](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=vsch)** commented

I encountered a show stopper. Doxia requires Java6 compatibility and flexmark-java requires Java7.

I converted the source to Java6 but BitSet in JDK7 is missing some BitSet methods. Additionally,  org.nibor.autolink used by the autolink extension is byte code level 7 not 6.

The missing BitSet methods I can get around but the byte code level of the autolink dependency would require rewriting the extension without the library.

Considering that JDK6 has reached EOL in 2012 is there a good reason why Doxia is still requiring JDK6 compatibility?
